src/test/java/com/hoteldetails/pad/service/RoomAvailabilityServerServiceTest.java

```diff
@@ -1,35 +1,66 @@
-class RoomAvailabilityServerServiceTest {
-
+@RunWith(MockitoJUnitRunner.class)
+public class RoomAvailabilityServerServiceTest {
     @Mock
     RoomAvailabilityService roomAvailabilityService;
+    @Mock
+    ErrorResponseMapper errorResponseMapper;
 
     @InjectMocks
     private RoomAvailabilityServerService roomAvailabilityServerService;
 
     @Test
-    void getRoomAvailability() {
-        RoomAvailabilityRequest.Builder request = RoomAvailabilityRequest.newBuilder();
+    public void getRoomAvailability() {
+        RoomAvailabilityRequest request = RoomAvailabilityRequest.newBuilder()
+                .setRequestContext(Context.newBuilder().setSupplierUrl("http://traveldoo.com").build()).build();
+        RoomAvailabilityResponse response = RoomAvailabilityResponse.newBuilder()
+                .setResponseStatus(ResponseStatus.newBuilder().setStatus(1).build()).build();
+        StreamRecorder<RoomAvailabilityResponse> responseObserver = StreamRecorder.create();
+        when(roomAvailabilityService.getRoomAvailabilityFromSupplier(request)).thenReturn(response);
+        roomAvailabilityServerService.getRoomAvailability(request, responseObserver);
+        List<RoomAvailabilityResponse> responseList = responseObserver.getValues();
+        RoomAvailabilityResponse roomAvailabilityResponse = responseList.get(0);
+        assertThat(responseList).isNotEmpty();
+        assertThat(roomAvailabilityResponse).isNotNull();
+        assertThat(roomAvailabilityResponse.getResponseStatus().getStatus()).isEqualTo(1);
+        verify(roomAvailabilityService, atLeast(1)).getRoomAvailabilityFromSupplier(request);
+    }
+
+    @Test
+    public void testForException(){
+        RoomAvailabilityRequest request = RoomAvailabilityRequest.newBuilder()
+                .setRequestContext(Context.newBuilder().setSupplierUrl("http://traveldoo.com").build()).build();
         StreamRecorder<RoomAvailabilityResponse> responseObserver = StreamRecorder.create();
-        roomAvailabilityServerService.getRoomAvailability(request.build(), responseObserver);
+        ResponseStatus.Builder reponseStatusBuilder = ResponseStatus.newBuilder();
+        when(roomAvailabilityService.getRoomAvailabilityFromSupplier(request)).thenThrow(HotelException.class);
+        lenient(). when(errorResponseMapper.mapErrorResponse(anyString(),anyString())).thenReturn(reponseStatusBuilder.build());
+        roomAvailabilityServerService.getRoomAvailability(request, responseObserver);
         List<RoomAvailabilityResponse> responseList = responseObserver.getValues();
         RoomAvailabilityResponse roomAvailabilityResponse = responseList.get(0);
         assertThat(responseList).isNotEmpty();
-        assertThat(roomAvailabilityResponse).isNull();
+        assertThat(roomAvailabilityResponse).isNotNull();
+        verify(roomAvailabilityService, atLeast(1)).getRoomAvailabilityFromSupplier(request);
+        verify(errorResponseMapper, atLeast(1)).mapErrorResponse(null,null);
```

Comment:
instaed of null, it should be anystring right? as inside when we are passing any string